### PR TITLE
update wagmi & rainbow for walletConnectV2

### PR DESCRIPTION
**Pull request by technophile-04**
## Description
Updated wagmi and rainbowkit there were close to no breaking changes and were only related to WalletConnect in both wagmi and rainbowkit checkout -> [wagmi migration from 0.11.x -> 0.12.x](https://0.12.x.wagmi.sh/react/migration-guide#012x-breaking-changes) and [rainbow kit migration guide](https://www.rainbowkit.com/docs/migration-guide#012x-breaking-changes)

**Demo** : 

https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/fedafa28-d22c-4780-9397-1d70fe2c0a95

**TODO** : 
- [x] We will be needing the `projectId` for the demo purpose ^ I created a test projectId

## Additional Information

- [X] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [X] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

None

Your ENS/address: shivbhonde.eth

### CODE REVIEW

File: packages/nextjs/services/web3/wagmiConnectors.tsx [hunk 1]
```diff
@@ -45,13 +45,14 @@ export const appChains = configureChains(
   },
 );
 
+const walletsOptions = { chains: appChains.chains, projectId: "" };
```
Comment (technophile-04) [inline packages/nextjs/services/web3/wagmiConnectors.tsx:48]: We need to add `projectId`

Comment (sverps) [reply packages/nextjs/services/web3/wagmiConnectors.tsx:48]: Would it make sense to move this `projectId` to the scaffold.config file and document it in the readme?

Comment (technophile-04) [reply packages/nextjs/services/web3/wagmiConnectors.tsx:48]: Yes !!! I Should have done it already, Tysm 🙌 will update it

Comment (technophile-04) [reply packages/nextjs/services/web3/wagmiConnectors.tsx]: Confused about whether we should add it to README or not, I have added a similar description to Alchemy API key, above it

File: packages/nextjs/scaffold.config.ts [hunk 2]
```diff
@@ -25,6 +26,12 @@ const scaffoldConfig = {
   // .env.local for local testing, and in the Vercel/system env config for live apps.
   alchemyApiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY || "oKxs-03sij-U_N0iOlrSsZFr29-IqbuF",
 
+  // This is ours WalletConnect's default project ID.
+  // You can get your own by at https://cloud.walletconnect.com
```
Comment (carletex) [inline packages/nextjs/scaffold.config.ts:30]:
```suggestion
  // You can get your own at https://cloud.walletconnect.com
```

File: packages/nextjs/services/web3/wagmiConnectors.tsx (hunk 1 above)
Comment (carletex) [reply packages/nextjs/services/web3/wagmiConnectors.tsx]: I think we should have a section in the README talking about the `scaffold.config.ts` file. Maybe in another issue + PR?

**Review by sverps — APPROVED** [review]

**Review by carletex — CHANGES_REQUESTED** [review]
I deployed a contract to Sepolia, changed the scaffold.config to point to it, and got this:

image

**Comment by technophile-04** [comment]
> I deployed a contract to Sepolia, changed the scaffold.config to point to it, and got this: 

Yes, it happened with me too !! For some reason, it works on goerli...I am having a look at it

**Review by sverps — APPROVED** [review]

**Comment by technophile-04** [comment]
> Ethers v5 doesn't support Sepolia yet. Will need to use the jsonRpcProvider as a workaround and enter the URL manually.

Yeah ! I am confused too, there was something in [`0.12.0`](https://github.com/wagmi-dev/wagmi/releases/tag/wagmi%400.12.0) that lead it to break (my assumption since it was not breaking at 0.11.7), I have asked the question here about it https://github.com/wagmi-dev/wagmi/issues/2219#issuecomment-1600285597 . 

Btw while testing I noticed till now weren't actually using our RPC when DApps were deployed to `sepolia` it was always defaulted to `rpc.sepolia.org`(public provider) ....checkout `networks` tabs of ->  https://sepolia-test.vercel.app

Maybe we can wait a few days until more `rainbowkit` patches comes out and hopefully will get an answer for this https://github.com/wagmi-dev/wagmi/issues/2219#issuecomment-1600285597 too 🙌

**Comment by carletex** [comment]
> Btw while testing I noticed till now weren't actually using our RPC when DApps were deployed to sepolia it was always defaulted to rpc.sepolia.org(public provider) ....checkout networks tabs of -> https://sepolia-test.vercel.app/

Oh, this makes sense. As you said in the Wagmi issue, it seems that 0.11 has a fallback that was used when the first provider 
 failed.

> Maybe we can wait a few days until more rainbowkit patches comes out and hopefully will get an answer for this https://github.com/wagmi-dev/wagmi/issues/2219#issuecomment-1600285597 too raised_hands

Your call!

**Comment by technophile-04** [comment]
Hey guys, merging this 

> RainbowKit enabled WalletConnect v2 by default with 0.12.15 and 1.0.2. We are continuing support for 0.12.x in case your dApp has not yet upgraded to wagmi v1 and RainbowKit v1.

Original post -> https://www.rainbowkit.com/guides/walletconnect-v2 

I think we should get #352 rollin as soon though

**Pull request merged**